src/components/input/Input.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { FC, forwardRef, ForwardedRef } from "react";
+import { StyledInput, Label } from "./Input.styled";
+
+export interface Props {
+  value: string;
+  // Note: updating the value needs to happen in this onChange function.
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  ref?: ForwardedRef<HTMLInputElement | null>;
+  label?: string;
+  placeholder?: string;
+  labelPlacement?: LabelPlacement;
+  width?: string;
+  onKeyDown?: React.KeyboardEventHandler<HTMLInputElement> | undefined;
+  id?: string;
+}
+
+export type LabelPlacement = "default" | "overlap";
+
+const _Input: FC<Props> = ({
+  value,
+  onChange,
+  label,
+  labelPlacement = "default",
+  width,
+  onKeyDown,
+  id,
+  placeholder,
+  ref = null,
```

Comment:
i liked the `...delegated` thing you did with the buttons, is there a reason not to do it here?

Reply:
Sure, we can spread over props if the consumer wants.